### ISSUE TITLE
Share usage example needs modification

### ISSUE THREAD

**Issue by jwrosewell** [51Degrees/device-detection-java#5]
Based on the information on this page http://51degrees.com/documentation/4.2/_examples__usage_sharing.html the following changes are needed.

1. The following code results in an error because the message size is larger than the default queue size.

 ``` Java
       ShareUsageElement usageElement = new ShareUsageBuilder(loggerFactory, http)
                .setSharePercentage(0.01)
                .setMinimumEntriesPerMessage(2500)
                .build();
```

it needs to be modified to become the following

``` Java
        ShareUsageElement usageElement = new ShareUsageBuilder(loggerFactory, http)
                .setSharePercentage(0.01)
                .setMaximumQueueSize(10000)
                .setMinimumEntriesPerMessage(2500)
                .build();
```

2. The links to GitHub in the example don't work.

3. The example doesn't differentiate between the implementation for a) building from configuration; and b) building in code. I think these should be two separate example code snippets so they're easier to follow.

**Comment by Steve51D** [comment]
With the release of 4.3.0 issues 1 and 2 above will be resolved.
In that version, `setMaximumQueueSize` is marked as deprecated and the value is automatically calculated based on the value supplied to `setMinimumEntriesPerMessage`

The link to GitHub doesn't work because the code for this example is not yet released to GitHub.

**Comment by Steve51D** [comment]
Verified that issues 1 and 2 are resolved with the 4.3.0 release as described above.
